src/menu/menu_functions.rs

```diff
@@ -145,12 +145,24 @@ pub fn parse_selection_char(buffer: &str, marker: char) -> ParseResult {
 pub fn find_common_string(values: &[Suggestion]) -> (Option<&Suggestion>, Option<usize>) {
     let first = values.iter().next();
 
+    let mut new_index: Option<usize> = match first {
+        Some(first) => Some(first.value.len()),
+        None => Some(0),
+    };
+
+    if values.len() <= 1 {
+        return (first, new_index);
+    }
```

Comment:
Maybe worth adding a comment what the different cases are. We have this early return and the chain from hell below.